src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -575,11 +575,46 @@ private void processPackage(DetailAST ast, DetailAST nextToken) {
                 log(ast, MSG_SHOULD_BE_SEPARATED, ast.getText());
             }
         }
-        if (!hasEmptyLineAfter(ast)) {
+        if (isLineEmptyAfterPackage(ast)) {
+            final DetailAST elementAST = getViolationASTForPackage(ast);
```

Comment:
`getViolationAstForPackage`

`elementAst`

please update all other cases with AST to Ast that you introduced in this PR.

Reply:
Done